### PR TITLE
Paged Table Template resolvers

### PR DESCRIPTION
**Pull request by Allcharles**
# Paged Table Template resolvers

Updated `PagedTableTemplate` to handle extracting models from resolvers and handling errors. Mostly this is an interim step before implementing the `MyProjectsComponent` which required these adjustments to `pagedTableTemplate`.

## Changes

- Created `retrieveResolvers` function
- Moved `getPageData()` into `ngOnInit()` function of `pagedTableTemplate`
- Created placeholder files for user project pages (will complete in another PR)

### CODE REVIEW

File: src/app/component/profile/pages/projects/my-projects.component.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  constructor(api: ProjectsService, route: ActivatedRoute) {
+    super(
+      api,
+      (projects) =>
+        projects.map((project) => ({
+          name: project.name,
+          sites: project.siteIds.size,
+          permission: "UNKNOWN",
+        })),
```
Comment (atruskie) [inline src/app/component/profile/pages/projects/my-projects.component.ts]: not clear from looking at code, but why not map to  `Project` model here?

**Review by atruskie — APPROVED** [review]
This PR seems like its mainly moving stuff around. TBH I don't get it, but I don't see anything wrong either.

If you have any specific questions let me know, otherwise LGTM

**Comment by Allcharles** [comment]
Entire PR basically boils down to the following lines: https://github.com/QutEcoacoustics/workbench-client/blob/c32b2b7ce7f29799a74938faaac9732dc9d0a738/src/app/helpers/tableTemplate/pagedTableTemplate.ts#L77-L89

File: src/app/component/profile/pages/projects/my-projects.component.ts (hunk 1 above)
Comment (Allcharles) [reply src/app/component/profile/pages/projects/my-projects.component.ts]: Ill have to fix this after the model cleanup PR #190

**Pull request merged**